crates/providers/src/new.rs

```diff
@@ -96,7 +130,87 @@ pub trait Provider<N: Network, T: Transport + Clone = BoxTransport>: Send + Sync
     /// Returns a [`Weak`] RPC client used to send requests.
     fn weak_client(&self) -> WeakClient<T>;
 
-    async fn new_pending_transaction(&self, tx_hash: B256) -> TransportResult<PendingTransaction>;
+    /// Watch for the confirmation of a single pending transaction with the given configuration.
+    ///
+    /// Note that this is handled internally rather than calling any specific RPC method.
+    async fn watch_pending_transaction(
+        &self,
+        config: PendingTransactionConfigInner,
+    ) -> TransportResult<PendingTransaction>;
```

Comment:
This is the internal method used by the `PendingTransactionConfig` type which uses the internal heartbeat